sov-modules/sov-state/src/scratchpad.rs

```diff
@@ -0,0 +1,267 @@
+use std::{cell::RefCell, fmt::Debug, ops::DerefMut, rc::Rc};
+
+use crate::{
+    internal_cache::StorageInternalCache,
+    storage::{StorageKey, StorageValue},
+    Storage,
+};
+use first_read_last_write_cache::cache::CacheLog;
+use sovereign_sdk::core::traits::Witness;
+
+/// A working set accumulates reads and writes on top of the underlying DB,
+/// automating witness creation.
+pub struct Delta<S: Storage> {
+    inner: S,
+    witness: Rc<S::Witness>,
+    cache: Rc<RefCell<StorageInternalCache>>,
+}
+
+/// A wrapper that adds additional reads and writes on top of an underlying Delta.
+/// These are handly for implementing operations that might revert on top of an existing
+/// working set, without running the risk that the whole working set will be discarded if some particular
+/// operation reverts.
+pub struct RevertableDelta<S: Storage> {
+    inner: Rc<RefCell<Option<Delta<S>>>>,
+    witness: Rc<S::Witness>,
+    cache: Rc<RefCell<StorageInternalCache>>,
+}
+
```

Comment:
In the `Witness` we are using a `RefCell `, If we expose it here and move all the `Rc<RefCell<..>>` one level higher and wrap the whole `RevertableDelta => Rc<RefCell< RevertableDelta >>` will it reduce the amount of `borrow & borrow_mut`?